code/game/turfs/open/floor/plating/misc_plating.dm

```diff
@@ -271,13 +271,37 @@
 /turf/open/floor/plating/grass
 	name = "grass"
 	desc = "A patch of grass."
-	icon_state = "grass"
-	broken_states = list("sand")
+	icon_state = "grass0"
+	var/smooth_icon = 'icons/turf/floors/grass.dmi'
 	bullet_bounce_sound = null
 	footstep = FOOTSTEP_GRASS
 	barefootstep = FOOTSTEP_GRASS
 	clawfootstep = FOOTSTEP_GRASS
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	smoothing_flags = SMOOTH_CORNERS
+	smoothing_groups = list(SMOOTH_GROUP_TURF_OPEN,SMOOTH_GROUP_FLOOR_GRASS)
+	canSmoothWith = list(SMOOTH_GROUP_CLOSED_TURFS,SMOOTH_GROUP_FLOOR_GRASS)
+	layer = HIGH_TURF_LAYER
+
+/turf/open/floor/plating/grass/Initialize()
+	if(smoothing_flags)
+		var/matrix/M = new
+		M.Translate(-9, -9)
+		transform = M
+		icon = smooth_icon
+	. = ..()
```

Comment:
```suggestion
	. = ..()
	if(smoothing_flags)
		var/matrix/translation = new
		translation.Translate(-9, -9)
		transform = translation
		icon = smooth_icon
```
* Always call parent first on `Initialize()` when you can get away with it.
* Use better variable names.